src/wp-includes/media.php

```diff
@@ -1965,17 +1972,70 @@ function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
 	 */
 	$optimization_attrs = wp_get_loading_optimization_attributes(
 		'img',
-		array(
-			'width'         => $width,
-			'height'        => $height,
-			'loading'       => $loading_val,
-			'fetchpriority' => $fetchpriority_val,
-		),
+		$args,
 		$context
 	);
 
-	// Images should have source and dimension attributes for the loading optimization attributes to be added.
-	if ( ! str_contains( $image, ' src="' ) || ! str_contains( $image, ' width="' ) || ! str_contains( $image, ' height="' ) ) {
+	// Images should have source for the decoding and the loading optimization attributes to be added.
```

Comment:
All three can be considered loading optimization attributes, so we can simplify this annotation.
```suggestion
	// Images should have source for the loading optimization attributes to be added.
```